### PR TITLE
Grouped multi selection prompt

### PR DESCRIPTION
**Pull request by DanielmTheDev**
@EluciusFTW Have a look at this. Not so sure about the solution as it seems a little clunky, especially regarding those group labels. 
Maybe you have some idea on how to improve upon it. 

image

### CODE REVIEW

File: src/spectrecoff/Spectre/Prompt.fs [hunk 1]
```diff
@@ -1,37 +1,59 @@
...
     { Secret: bool
       Optional: bool }
 
-type MultiSelectionPromptOptions = 
+type MultiSelectionPromptOptions =
     { PageSize: int }
 
-let mutable defaultOptions = 
+type ChoiceGroup<'T> =
+    { Group: 'T
```
Comment (EluciusFTW) [inline src/spectrecoff/Spectre/Prompt.fs:14]: Why does the group have to be a 'T? Isn't it just a title and not an item in of itself?

File: src/spectrecoff/Spectre/Prompt.fs [hunk 2]
```diff
@@ -1,37 +1,59 @@
...
         let prompt = MultiSelectionPrompt()
         prompt.AddChoices (choices |> Seq.toArray) |> ignore
         prompt.Title <- question
         prompt.PageSize <- options.PageSize
         prompt
 
-    let textPrompt<'T> question (options: PromptOptions) = 
+    let groupedMultiSelectionPrompt<'T> options question (choiceGroups: ChoiceGroups<'T>) =
+        choiceGroups.Groups
+        |> Seq.fold (fun (prompt: MultiSelectionPrompt<'T>) groups -> prompt.AddChoiceGroup<'T>(groups.Group, groups.Choices)) (MultiSelectionPrompt())
```
Comment (EluciusFTW) [inline src/spectrecoff/Spectre/Prompt.fs:49]: This variable should be `group`, imo.

File: src/spectrecoff/Spectre/Prompt.fs [hunk 3]
```diff
@@ -44,26 +66,32 @@ module private Prompts =
...
-let chooseMultipleFromWith options (choices: string list) question = 
+let chooseMultipleFromWith options (choices: string list) question =
     prompt (Prompts.multiSelectionPrompt question choices options)
 
-let chooseMultipleFrom = 
+let chooseMultipleFrom =
     chooseMultipleFromWith defaultMultiSelectionOptions
 
-let ask<'T> question = 
+let chooseMultipleGroupedFromWith<'T> options (groupedChoices: ChoiceGroups<'T>) question =
```
Comment (EluciusFTW) [inline src/spectrecoff/Spectre/Prompt.fs:78]: The only naming idea I have is to drop the _Multiple_ part, as you would not use groups if you only allow to select a single item. I'd then also switch the order of the words to `chooseFromGroupedWith`.

**Review by EluciusFTW — APPROVED** [review]
Sweet feature, works like a charm. Thanks!

File: src/spectrecoff/Spectre/Prompt.fs (hunk 1 above)
Comment (DanielmTheDev) [reply src/spectrecoff/Spectre/Prompt.fs:14]: I found this confusing too, and mentioned it in the documentation. This is how it is in the Spectre api. 
In any case I was planning to approach you about a better solution, stay tuned.

File: src/spectrecoff/Spectre/Prompt.fs (hunk 2 above)
Comment (DanielmTheDev) [reply src/spectrecoff/Spectre/Prompt.fs]: agreed, thanks

File: src/spectrecoff/Spectre/Prompt.fs (hunk 3 above)
Comment (DanielmTheDev) [reply src/spectrecoff/Spectre/Prompt.fs]: You make a good point, done

**Pull request merged**